orderbook/src/fee.rs

```diff
@@ -276,13 +278,17 @@ impl MinFeeCalculating for MinFeeCalculator {
             .await
         {
             if fee >= apply_fee_factor(past_fee, app_based_fee_factor) {
```

Comment:
I think `past_fee` already has the "base" fee factor applied. Maybe we should refactor the fee measurement to already capture completely un-subsidized fee.

Then, instead of
```rust
        let app_based_fee_factor = *self
            .partner_additional_fee_factors
            .get(&app_data)
            .unwrap_or(&1.0);
```

we could:
```rust
        let fee_factor = *self
            .partner_additional_fee_factors
            .get(&app_data)
            .unwrap_or(&1.0) * self.fee_factor;
```

This would make the `full_fee_amount` be the actual full fee amount, make changes to the "base" fee factor not affect orders (as the full amount will be stored, instead of full amount times "base" fee factor) and allow us to remove the `--fee-factor` configuration from the `solver` binary as well.

Reply:
Nice catch! Yeah, sure I'll implement this.